000-cbdb-sandbox/configs/smoke-test.sh

```diff
@@ -0,0 +1,216 @@
+
+    CREATE_LILY=$(psql -c "\du" | grep lily)
+
+    if [[ -z $CREATE_LILY ]]; then
+        echo "Create user Lily failed"
+        exit 1
+    fi
+
+    createuser --interactive lucy -s
+
+    CREATE_LUCY=$(psql -c "\du" | grep lucy)
+
+    if [[ -z $CREATE_LUCY ]]; then
+        echo "Create user Lily failed"
+        exit 1
+    fi
+
+    psql -c "CREATE ROLE users"
+
+    psql -c "GRANT users TO lily, lucy"
+
+    LUCY_USERS=$(psql -c "\du" | grep lucy | grep "{users}")
+    LILY_USERS=$(psql -c "\du" | grep lily | grep "{users}")
+
+    if [[ -z $LUCY_USERS || -z $LILY_USERS ]]; then
+        echo "Create user Lily failed"
+        exit 1
+    fi
+
+    LILY_LOGIN_FAIL=$(psql -U lily -d gpadmin -c "SELECT version()" 2>&1 || true)
+
+    if [[ $LILY_LOGIN_FAIL != *"no pg_hba.conf entry for host"* ]]; then
+        echo "Command failed"
+        exit 1
+    fi
+
+    LUCY_LOGIN_FAIL=$(psql -U lucy -d gpadmin -c "SELECT version()" 2>&1 || true)
+
+    if [[ $LUCY_LOGIN_FAIL != *"no pg_hba.conf entry for host"* ]]; then
+        echo "Command failed"
+        exit 1
+    fi
+
+    echo "local gpadmin lily md5" >> /data0/database/master/gpseg-1/pg_hba.conf
+    echo "local gpadmin lucy trust" >> /data0/database/master/gpseg-1/pg_hba.conf
+
+    gpstop -u
+
+    PGPASSWORD=changeme psql -d gpadmin -U lily -c 'SELECT VERSION()'
+
+    psql -d gpadmin -U lucy -c 'SELECT VERSION()'
+}
+
+lesson2(){
+    echo "################################"
+    echo "         LESSON 2 TESTS         "
+    echo "################################"
+
+    DROPDB=$(dropdb $TUTORIALDB 2>&1 || true)
+    
+    if [[ $DROPDB != *"database \"$TUTORIALDB\" does not exist"* ]]; then
+        echo "Command failed"
+        exit 1
+    fi
+
+    createdb $TUTORIALDB
+
+    CREATEDB=$(psql -l | grep $TUTORIALDB)
+    
+    if [[ -z $CREATEDB ]]; then
+        echo "Database $TUTORIALDB not found, exiting"
+        exit 1
+    fi
+
+    echo "local $TUTORIALDB lily md5" >> /data0/database/master/gpseg-1/pg_hba.conf
+
+    gpstop -u
+
+    PGPASSWORD=changeme psql -d $TUTORIALDB -U lily -c 'SELECT VERSION()'
+
+    psql -U gpadmin -d tutorial -c "GRANT ALL PRIVILEGES ON DATABASE tutorial TO lily"
+
+    PGPASSWORD=changeme psql -d $TUTORIALDB -U lily -c "DROP SCHEMA IF EXISTS faa CASCADE"
+    PGPASSWORD=changeme psql -d $TUTORIALDB -U lily -c "CREATE SCHEMA faa"
+    PGPASSWORD=changeme psql -d $TUTORIALDB -U lily -c "ALTER ROLE lily SET search_path TO faa, public, pg_catalog, gp_toolkit"
+    LILY_SEARCH_PATH=$(PGPASSWORD=changeme psql -d $TUTORIALDB -U lily -c "SHOW search_path")
+
+    if [[ $LILY_SEARCH_PATH != *"faa, public, pg_catalog, gp_toolkit"* ]]; then
+        echo "Lily search path not set correctly, exiting"
+        exit 1
+    fi 
+}
+
+lesson3(){
+    echo "################################"
+    echo "         LESSON 3 TESTS         "
+    echo "################################"
+
+    cd /tmp
+    tar xzf faa.tar.gz
```

Comment:
In my environment, the file /tmp/faa.tar.gz was not readable by gpadmin and thus the tar failed.

```
-rw-r----- 1 root    root    63088802 Aug  6 08:25 faa.tar.gz
```

Reply:
Strange, it's owned by root for me as well however as gpadmin I can still untar it. Changed the init_system script to make gpadmin the owner just to be safe
https://github.com/antoniopetrole/bootcamp/commit/51c5727a455b85f60c2697ee504f6d11b2b645ea